### DISCUSSION TITLE
Airflow scheduler stops receiving heartbeat all of a sudden multiple times

### DISCUSSION THREAD

**Discussion by mangesh-ux** [apache/airflow#35565] [Q&A]
I have an airflow instance hosted on EC2 server with 4GB RAM, I am using a remote pgsql db for metadata. It was working seamless until 4-5 months but suddenly I have started to receive the message that "The scheduler does not appear to be running. Last heartbeat was received X minutes ago". Nothing was found ion scheduler logs. I have to restart airflow scheduler every time to solve this but it reappears. Any help would be greatly appreciated.

**Comment by potiuk** [comment]
See what is the root cause of it in the logs - you should see some logs. Maybe experiment with increasing memory 4GB is not a lot for running airflow instance and generally memory usage depends on the type and nature of DAGs you have so it might change over time.

**Comment by rdjouder** [comment]
Hello mangesh-ux, my general question is on the same issue:
Airflow 2.7.2 /health endpoint returns scheduler unhealthy but schedulers are perfectly fine.
My setup is similar to yours. I am pretty sure that my schedulers OK.
Any one can describe the heartbeat feature ? Does Scheduler regularly send post requests to airflow node, or is it communication through DB ?

**Reply by mangesh-ux** [reply]
@rdjouder how do you start the scheduler? and does restarting the scheduler fix this problem for you?

**Reply by rdjouder** [reply]
exec airflow scheduler

**Reply by mangesh-ux** [reply]
try other way like systemd or nohup. This was causing me problems and if restarting solves the problem, you can just write a script to check health endpoint and if it is unhealthy, write a script to start scheduler programmatically using subprocess. you can schedule this health-check script using cron.

**Reply by potiuk** [reply]
@rdjouder -> maybe [Documentation](https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/logging-monitoring/check-health.html) might help ? It's a wise idea to try to search the official documentation before asking humans to do what is already done in the docs, I think. Then (as explained at the beginning) after you understnd what you can do with checking health and looking at the way how it is done in YOUR deployment you can take a look at the logs and see the root cause of the problem.
